# Test info

- Name: Login Functionality with Multiple Valid Users >> should login successfully with username: visual_user
- Location: C:\Senior-test-engineer-assessment\tests\login.spec.ts:7:9

# Error details

```
Error: page.goto: Test timeout of 30000ms exceeded.
Call log:
  - navigating to "https://www.saucedemo.com/", waiting until "load"

    at LoginPage.goto (C:\Senior-test-engineer-assessment\pages\LoginPage.ts:11:21)
    at C:\Senior-test-engineer-assessment\tests\login.spec.ts:9:23
```

# Test source

```ts
   1 | import { Page, expect } from '@playwright/test';
   2 |
   3 | export class LoginPage {
   4 |   readonly page: Page;
   5 |
   6 |   constructor(page: Page) {
   7 |     this.page = page;
   8 |   }
   9 |
  10 |   async goto() {
> 11 |     await this.page.goto('https://www.saucedemo.com/');
     |                     ^ Error: page.goto: Test timeout of 30000ms exceeded.
  12 |   }
  13 |
  14 |   async login(username: string, password: string) {
  15 |     await this.page.fill('#user-name', username);
  16 |     await this.page.fill('#password', password);
  17 |     await this.page.click('.btn_action');
  18 |   }
  19 |
  20 |   async assertLoginSuccess() {
  21 |     await expect(this.page).toHaveURL(/.*inventory.html/);
  22 |     await expect(this.page.locator('.title')).toHaveText(/Products/i);
  23 |   }
  24 | }
  25 |
```